src/Microsoft.ML.Api/InternalSchemaDefinition.cs

```diff
@@ -131,19 +131,29 @@ private InternalSchemaDefinition(Column[] columns)
         }
 
         /// <summary>
-        /// Given a field info on a type, returns whether this appears to be a vector type,
+        /// Given a field or property info on a type, returns whether this appears to be a vector type,
         /// and also the associated data kind for this type. If a data kind could not
         /// be determined, this will throw.
         /// </summary>
-        /// <param name="fieldInfo">The field info to inspect.</param>
+        /// <param name="memberInfo">The field or property info to inspect.</param>
         /// <param name="isVector">Whether this appears to be a vector type.</param>
         /// <param name="kind">The data kind of the type, or items of this type if vector.</param>
-        public static void GetVectorAndKind(FieldInfo fieldInfo, out bool isVector, out DataKind kind)
+        public static void GetVectorAndKind(MemberInfo memberInfo, out bool isVector, out DataKind kind)
         {
-            Contracts.AssertValue(fieldInfo);
-            Type rawFieldType = fieldInfo.FieldType;
-            var name = fieldInfo.Name;
-            GetVectorAndKind(rawFieldType, name, out isVector, out kind);
+            Contracts.AssertValue(memberInfo);
+            switch (memberInfo)
+            {
+                case FieldInfo fieldInfo:
+                    GetVectorAndKind(fieldInfo.FieldType, fieldInfo.Name, out isVector, out kind);
+                    break;
+
+                case PropertyInfo propertyInfo:
+                    GetVectorAndKind(propertyInfo.PropertyType, propertyInfo.Name, out isVector, out kind);
+                    break;
+
+                default:
+                    throw Contracts.ExceptNotSupp("Expected a FieldInfo or a PropertyInfo");
```

Comment:
>throw [](start = 20, length = 5)

Same here: if the code is never expected to execute, make it obvious to reader by putting Contracts.Assert(false) ahead of throw.